app/src/main/java/com/example/wallet/MainActivity.kt

```diff
@@ -26,6 +29,28 @@ class MainActivity : AppCompatActivity() {
         setCurrentFragment(firstFragment)
 
         val bottomNavigationView: BottomNavigationView = findViewById(R.id.bottomNavigationView)
+        bottomNavigationView.setOnItemReselectedListener { item ->
+            when (item.itemId) {
+                R.id.homeFragment -> {
+                if(firstFragment.isAdded() && firstFragment.isVisible()){
```

Comment:
Please define variable name to be specific. 
Example: 'firstFragment' -> 'homeFragment'.
'thirdFragment' -> 'chartFragment'.